src/components/PrettifyQuery.tsx

```diff
@@ -26,10 +37,21 @@ const PrettifyQuery: FC<Props> = ({
   const handleClickPrettify = async () => {
     setLoading(true)
     try {
-      const response = await datasource.prettifyRequest(query.expr)
+      let { expr } = query;
```

Comment:
Could you please add tests for this?
Just want to verify that conversion from original query to modified and back is correct.

Reply:
The logic was improved, and tests were added. 
In the test, I used a mock response. It is not prettified, just normal functions. 
But all tests shows that grafana variables correctly removed from the the request with default lookbehind window and after it correctly replaced with grafana variable

Reply:
I'm not sure if such requests can be used, but they can be executed without errors.
The logic works incorrectly:
1) If there are multiple variables in the expression,
2) If the expression includes `[1i]`.

````javascript
   rate(metric_name{mode="idle"}[$__interval]) + up{instance="[1i]"} 
// rate(metric_name{mode="idle"}[$__interval]) + up{instance="[$__interval]"}

   rate(metric_name[$__interval]) + rate(metric_name[$__range]) 
// rate(metric_name[$__range]) + rate(metric_name[$__range])

````

Reply:
i think it is a valid points, will update tests and logic